packages/db/src/query/sensor.ts

```diff
@@ -628,8 +628,17 @@ export async function assignSensorToUser(clientId: string, userId: string | null
 /**
  * Delete the current user from the sensor without inserting it in sensorHistory
  */
-export async function deleteUserFromSensor(clientId: string) {
-    await db.update(sensor).set({ userId: null }).where(eq(sensor.clientId, clientId));
+export async function resetSensorUser(clientId: string) {
+    await db.transaction(async (trx) => {
+        const sensors = await trx.select({ id: sensor.id }).from(sensor).where(eq(sensor.clientId, clientId));
+        if (sensors.length === 0) {
+            return;
+        }
+
+        await trx.update(sensor).set({ userId: null }).where(eq(sensor.clientId, clientId));
+        await trx.delete(sensorToken).where(eq(sensorToken.sensorId, sensors[0].id));
+        await trx.delete(sensorData).where(eq(sensorData.sensorId, sensors[0].id));
```

Comment:
Wir hatten doch erst geplant, dass ein Benutzer seine Messwerte auch dann noch einsehen kann, wenn ihm der Sensor gar nicht mehr zugewiesen ist. Das wäre ja so nicht möglich, wenn die Daten gelöscht werden

Reply:
Im Normalbetrieb sollten wir das auch über den UserSelect machen.

Die ist eher für uns als Entwickler, da z.B. Simon und ich oft die Sensoren gewechselt haben und dann mussten wir jedesmal den Token und auch die Werte zurückgesetzten.

Reply:
Also verstehe ich dich richtig, dass das Zurücksetzen über die Action-Cell anders ist als über den User-Select?

Reply:
Ja, über UserSelect, wird auch die Tabelle Sensor History gefüllt, dem Sensor eine neue ID gegeben damit er für den neuen Nutzer genutzt werden kann.

Nutzer zurücksetzen oder vllt. besser Sensor zurücksetzen, löscht den Nutzer vom Sensor sowie alle Token und Werte. Ist eher für die Entwicklung um schnell die Daten eines Sensors zu löschen.

Vllt. sollte da noch ein Popup mit einer weiteren Abfrage dann implementiert werden?

Reply:
Ja, den Popup finde ich in dem Fall auch sinnvoll